plugins/plugin-docker/che-plugin-openshift-client/src/main/java/org/eclipse/che/plugin/openshift/client/OpenShiftConnector.java

```diff
@@ -1431,57 +1362,49 @@ private void cleanUpWorkspaceResources(String deploymentName) throws IOException
         throw new OpenShiftException("Timeout while waiting for pods to terminate");
     }
 
-    private String getWorkspaceDir(String[] volumes) {
+    private void createWorkspaceDir(String[] volumes) throws OpenShiftException {
         PersistentVolumeClaim pvc = getClaimCheWorkspace();
-        if (pvc != null) {
-            for (String volume : volumes) {
-                String mountPath = volume.split(":",3)[1];
-                if (cheWorkspaceProjectsStorage.equals(mountPath)) {
-                    String hostPath = volume.split(":",3)[0];
-                    String subPath = hostPath.replace(cheWorkspaceStorage, "");
-                    if (subPath.startsWith("/")) {
-                        subPath = subPath.substring(1);
-                    }
-                    return mountPath + "/" + subPath;
+        String workspaceSubpath = getWorkspaceSubpath(volumes);
+        if (pvc != null && !isNullOrEmpty(workspaceSubpath)) {
+            LOG.info("Making sure directory exists for workspace {}", workspaceSubpath);
+            boolean succeeded = OpenShiftPvcHelper.createJobPod(workspacesPersistentVolumeClaim,
+                                                                openShiftCheProjectName,
+                                                                "create-",
+                                                                OpenShiftPvcHelper.Command.MAKE,
+                                                                workspaceSubpath);
+            if (!succeeded) {
+                LOG.error("Failed to create workspace directory {} in PVC {}", workspaceSubpath,
+                                                                               workspacesPersistentVolumeClaim);
+                throw new OpenShiftException("Failed to create workspace directory in PVC");
+            }
+        }
+    }
+
+    private String getWorkspaceSubpath(String[] volumes) {
+        String workspaceSubpath = null;
+        for (String volume : volumes) {
+            String mountPath = volume.split(":", 3)[1];
```

Comment:
It's a bit of a magic number for me too --  this method is mostly renamed and rearranged from the previous subpaths implementation. I've tried adding some documentation to explain this method but it still could probably be clearer.